apps/recnet/src/app/[handle]/Profile.tsx

```diff
@@ -195,6 +204,79 @@ function EditProfileDialog(props: { handle: string }) {
                 </Text>
               ) : null}
             </label>
+            <label>
+              <Text as="div" size="2" mb="1" weight="medium">
+                Personal Website
+              </Text>
+              <TextField.Root
+                placeholder="Personal website URL(Optional)"
+                {...register("url", {
+                  setValueAs: (val) => (val === "" ? null : val),
+                })}
+              />
+              {formState.errors.url ? (
+                <Text size="1" color="red">
+                  {formState.errors.url.message}
+                </Text>
+              ) : null}
```

Comment:
[nit] The Save button is disabled when the URL is invalid without an error message. I'll need to click somewhere else to see the validation message, which I feel like not very intuitive. Do you think not disabling the button and showing the message when the user clicks save would be a better idea?

Reply:
I think what you are describing is the validating mode `onSubmit` [here](https://react-hook-form.com/docs/useform#mode). I usually used `onTouched` since I think is a great middle ground between `onBlur` and `onChange`. 

`onBlur` is too insensitive since it will only validate (and show error message if needed) when you click elsewhere. On the other hand, `onChange` is too sensitive since it will start validating and show error message when you are in the middle of typing (bad experience I think). `onTouched` will validate at the very first `blur` event and after that, it will behave like `onChange`.

I think `onSubmit` would be a nice idea as well. The only tradeoff is you won't see any error message after you finish inputed one field. You will only get error message when click "Submit" button. (`onTouched` is better at this since user can get error message earlier.)

 Imo, choosing whether `onTouched` or `onSubmit` depends on when you want user to see the error messages, Both of them do great job at validating inputs and maintaining a good UX.

Reply:
Anyway, since it's not a critical issue and bug, I'm gonna ship this now. But feel free to discuss here or open another issue to discuss this.